x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix_vega_lite_spec.ts

```diff
@@ -0,0 +1,176 @@
+  if (outliers) {
+    return {
+      condition: {
+        value: COLOR_OUTLIER,
+        test: `(datum['${OUTLIER_SCORE_FIELD}'] >= mlOutlierScoreThreshold.cutoff)`,
+      },
+      value: COLOR_GRAY,
+    };
+  }
+
+  if (color !== undefined && legendType !== undefined) {
+    return {
+      field: color,
+      type: legendType,
+      scale: {
+        range: legendType === LEGEND_TYPES.NOMINAL ? COLOR_RANGE_NOMINAL : COLOR_RANGE_QUANTITATIVE,
+      },
+    };
+  }
+
+  return { value: DEFAULT_COLOR };
+};
+
+export const getScatterplotMatrixVegaLiteSpec = (
+  values: any[],
+  columns: string[],
+  resultsField?: string,
+  color?: string,
+  legendType?: LegendType,
+  dynamicSize?: boolean
+): TopLevelSpec => {
+  const outliers = resultsField !== undefined;
+  const transform = columns.map((column) => ({
+    calculate: `datum['${column}']`,
+    as: column,
+  }));
+  transform.push({
+    calculate: `datum['${resultsField}.${OUTLIER_SCORE_FIELD}']`,
+    as: OUTLIER_SCORE_FIELD,
+  });
+
+  return {
+    $schema: 'https://vega.github.io/schema/vega-lite/v4.8.1.json',
+    padding: 10,
+    config: {
+      axis: {
+        domainColor: '#ccc',
+        tickColor: '#ccc',
+        labelColor: '#aaa',
+        titleColor: '#999',
+      },
+    },
+    repeat: {
+      column: columns,
+      row: columns.slice().reverse(),
+    },
+    spec: {
+      data: { values },
+      mark: {
+        ...(outliers && dynamicSize
+          ? {
+              type: 'circle',
+              strokeWidth: 1.2,
+              strokeOpacity: 0.75,
+              fillOpacity: 0.1,
+            }
+          : { type: 'circle', opacity: 0.75, size: 8 }),
+      },
+      encoding: {
+        color: getColorSpec(outliers, color, legendType),
+        ...(dynamicSize
+          ? {
+              stroke: getColorSpec(outliers, color, legendType),
+              opacity: {
+                condition: {
+                  value: 1,
+                  test: `(datum['${OUTLIER_SCORE_FIELD}'] >= mlOutlierScoreThreshold.cutoff)`,
+                },
+                value: 0.5,
+              },
+            }
+          : {}),
+        ...(outliers
+          ? {
+              size: {
+                ...(!dynamicSize
+                  ? {
+                      condition: {
+                        value: 40,
+                        test: `(datum['${OUTLIER_SCORE_FIELD}'] >= mlOutlierScoreThreshold.cutoff)`,
+                      },
+                      value: 8,
+                    }
+                  : {
+                      type: LEGEND_TYPES.QUANTITATIVE,
+                      field: OUTLIER_SCORE_FIELD,
+                      scale: {
+                        type: 'linear',
+                        range: [8, 200],
+                        domain: [0, 1],
+                      },
+                    }),
+              },
+            }
+          : {}),
+        tooltip: { type: LEGEND_TYPES.QUANTITATIVE, field: OUTLIER_SCORE_FIELD },
```

Comment:
Is it possible to customize the text in the tooltip, to show e.g. `outlier score: 0.996`? The precision also seems overly high.

![image](https://user-images.githubusercontent.com/7405507/101790501-6bd09980-3afa-11eb-9abf-7201a9da514c.png)

Reply:
Improved tooltips to display all included fields and format outlier score up to 3 decimals in 63c260b9153.

<img src="https://user-images.githubusercontent.com/230104/102225366-723e8700-3ee7-11eb-9b79-879008b7a48b.png" width="40%" />